netsocadmin/routes/tools/account.py

```diff
@@ -0,0 +1,78 @@
+# stdlib
+from typing import Tuple
+
+# lib
+import flask
+import structlog as logging
+
+# local
+import login_tools
+import register_tools
+
+from .index import ProtectedToolView
+
+
+class AccountView(ProtectedToolView):
+    template_file = "account.html"
+
+    page_title = "Manage Account"
+
+    active = "account"
+
+    logger = logging.getLogger("netsocadmin.account")
+
+    def dispatch_request(self) -> str:
+        return self.render()
+
+
+class AbstractAccountView(AccountView):
+    """
+    Extend the ProtectedToolView with methods that help abstract some of the work out of the account related views
+    """
+    # Specify which method(s) are allowed to be used to access the route
+    methods = ["POST"]
+
+    def validate(self, username: str, password: str) -> Tuple[bool, str]:
+        """
+        Given a username and password validate them.
+            - Ensure the fields all have value
+            - Ensure the username / password combo is correct
+        :param username: The username to be validated
+        :param password: The password to be validated
+        """
+        # Check that all passed fields have value
+        fields = [username, password]
+        if not all(fields):
+            return False, "Please specify all fields."
+        # Check that the username / password combination is correct
+        login_user = login_tools.LoginUser(username, password)
+        if not login_tools.is_correct_password(login_user):
+            return False, f"Wrong username or password for user {username}."
+        # We good
+        return True, ""
```

Comment:
Probably be important if you checked that the passed username is equal to the current sessions username ;) i was able to change sofias password while logged in as john

Reply:
That's fair. You can also change other user's MySQL passwords at the moment so that should probably be patched too.

Reply:
Actually... why are they given the option to enter a username at all if that's the case?

Reply:
Probably just to confirm credentials. I *think* this was before my time